dotnet-interactive.Tests/HttpApiTests.cs

```diff
@@ -246,6 +248,72 @@ public async Task unimplemented_lsp_handlers_return_empty_string(Language langua
             responseJson.Should().BeEmpty();
         }
 
+        [Fact]
+        public async Task lsp_methods_run_deferred_commands()
+        {
+            // declare a variable in deferred code
+            var kernelBase = GetServer().Kernel as KernelBase;
+            kernelBase.DeferCommand(new SubmitCode("var one = 1;"));
+
+            // it's not defined
+            var response = await GetServer().HttpClient.GetAsync($"/variables/{Language.CSharp.LanguageName()}/one");
+            response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+
+            // ensure we can get http info about it
+            var encoded = Convert.ToBase64String(Encoding.UTF8.GetBytes("Console.WriteLine(one);"));
+            //                                                                              ^ (0, 20)
+            var request = new HoverParams()
+            {
+                TextDocument = new TextDocument() { Uri = $"data:text/plain;base64,{encoded}" },
+                Position = new Position()
+                {
+                    Line = 0,
+                    Character = 20,
+                },
+            };
+            response = await GetServer().HttpClient.PostJsonAsync("/lsp/textDocument/hover", request.SerializeLspObject());
+            await response.ShouldSucceed();
+            var responseJson = await response.Content.ReadAsStringAsync();
+            var hover = JsonConvert.DeserializeObject<TextDocumentHoverResponse>(responseJson);
+            hover.Contents.Value.Should().Be("(field) int one");
+
+            // and it now exists
+            response = await GetServer().HttpClient.GetAsync($"/variables/{Language.CSharp.LanguageName()}/one");
+            var responseContent = await response.Content.ReadAsStringAsync();
+            responseContent.Should().BeJsonEquivalentTo(1);
+        }
+
+        [Fact]
+        public async Task lsp_textDocument_hover_handles_growing_script_offsets()
```

Comment:
This kind of test should be done closer to the domain layer, as the thing being tested isn't the serialization or HTTP API.

Reply:
Will do.

Reply:
Check commit 'create separate LSP (de)serialization...` where I use the object model directly in the HTTP tests, and add separate JSON serialization/deserialization tests.